### PR TITLE
[MOBILE-3864] Add editTags to proxy

### PR DESCRIPTION
**Pull request by Ulrico972**
Tested on flutter

### CODE REVIEW

File: android/airship-framework-proxy/src/main/java/com/urbanairship/android/framework/proxy/TagOperation.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public data class TagOperation(
+    public val tags: List<String>,
+    public val action: TagOperationAction
+) {
+    public constructor(json: JsonMap) : this(
+        tags = json.require("tags").requireList().map { it.requireString() },
+        action = TagOperationAction.valueOf(
+            json.require("operationType").requireString().uppercase()
+        )
+    )
```
Comment (ilitvinenko) [inline android/airship-framework-proxy/src/main/java/com/urbanairship/android/framework/proxy/TagOperation.kt:20]: how do we handle exceptions? if json is invalid will it crash the app or just print an exception?

File: ios/AirshipFrameworkProxy/Proxies/AirshipChannelProxy.swift [hunk 2]
```diff
@@ -28,6 +28,24 @@ public class AirshipChannelProxy {
...
         }
     }
+    
+    @objc
+    public func editTags(json: Any) throws {
+        let data = try JSONUtils.data(json)
+        let operations = try JSONDecoder().decode(
+            [TagOperation].self,
+            from: data
+        )
```
Comment (ilitvinenko) [inline ios/AirshipFrameworkProxy/Proxies/AirshipChannelProxy.swift:38]: same here, what happens if we try to feed it with an invalid json?

Comment (rlepinski) [reply ios/AirshipFrameworkProxy/Proxies/AirshipChannelProxy.swift:38]: We bubble it up to the framework that uses it. Usually its a promise type API that we can provide the error

**Review by rlepinski — COMMENTED** [review]
Could we get a unit test on the JSON parsing?

File: android/airship-framework-proxy/src/main/java/com/urbanairship/android/framework/proxy/TagOperation.kt (hunk 1 above)
Comment (Ulrico972) [reply android/airship-framework-proxy/src/main/java/com/urbanairship/android/framework/proxy/TagOperation.kt:20]: The proxy methods are only used internally by our frameworks, then our frameworks should have some editors that wrap the values correctly so I think it should be ok

File: AirshipFrameworkProxy.podspec [hunk 3]
```diff
@@ -15,6 +15,6 @@ Pod::Spec.new do |s|
    s.requires_arc            = true
    s.swift_version           = "5.0"
    s.source_files            = "ios/AirshipFrameworkProxy/**/*.{h,m,swift}"
-   s.dependency                'Airship', "17.1.2"
+   s.dependency                'Airship', "17.2.1"
```
Comment (rlepinski) [inline AirshipFrameworkProxy.podspec:18]: 17.2.2

**Review by rlepinski — APPROVED** [review]

**Pull request merged**